### PR TITLE
Implement returndatacopy and returndatasize

### PR DESCRIPTION
**Pull request by nbanmp**
Also includes a few minor bugfixes

### CODE REVIEW

File: mythril/laser/ethereum/instructions.py [hunk 1]
```diff
@@ -1057,9 +1057,45 @@ def returndatacopy_(self, global_state: GlobalState) -> List[GlobalState]:
...
+        try:
+            concrete_memory_offset = helper.get_concrete_int(memory_offset)
+        except TypeError:
+            log.debug("Unsupported symbolic memory offset in RETURNDATACOPY")
+            return [global_state]
+
+        try:
+            concrete_return_offset = helper.get_concrete_int(return_offset)
+        except TypeError:
+            log.debug("Unsupported symbolic return offset in RETURNDATACOPY")
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: If we return [globalstate] for these catches, then we don't need the ifstatement at 1090. Additionally we won't need ln 1066, 1067 if we do that, as everything would always be initialized after the try catches

File: mythril/laser/ethereum/instructions.py [hunk 2]
```diff
@@ -1728,7 +1773,7 @@ def call_post(self, global_state: GlobalState) -> List[GlobalState]:
         instr = global_state.get_current_instruction()
 
         try:
-            callee_address, _, _, value, _, _, memory_out_offset, memory_out_size = get_call_parameters(
+            callee_address, callee_account, call_data, value, gas, memory_out_offset, memory_out_size = get_call_parameters(
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: why did you add this change?

File: mythril/laser/ethereum/instructions.py [hunk 3]
```diff
@@ -1694,14 +1739,14 @@ def call_(self, global_state: GlobalState) -> List[GlobalState]:
...
             return [global_state]
-        global_state.mstate.stack.append(
-            global_state.new_bitvec("retval_" + str(instr["address"]), 256)
-        )
 
         native_result = native_call(
             global_state, callee_address, call_data, memory_out_offset, memory_out_size
         )
         if native_result:
+            global_state.mstate.stack.append(
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: Don't we know the success/failure of the native call?

File: mythril/laser/ethereum/instructions.py [hunk 4]
```diff
@@ -1057,9 +1057,45 @@ def returndatacopy_(self, global_state: GlobalState) -> List[GlobalState]:
...
+            concrete_size is None
+            or global_state.last_return_data is None
+            or concrete_return_offset is None
+        ):
+            return [global_state]
+
+        global_state.mstate.mem_extend(concrete_memory_offset, concrete_size)
+        for i in range(concrete_size):
+            global_state.mstate.memory[concrete_memory_offset + i] = (
+                global_state.last_return_data[concrete_return_offset + i]
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: what if there is no last return data? returndatasize does have logic handling this

File: mythril/laser/ethereum/instructions.py (hunk 2 above)
Comment (norhh) [reply mythril/laser/ethereum/instructions.py]: there was an extra `_` there compared to the return statement of the get_call_parameters. Probably we shouldn't return too many parameters :sweat_smile:, like returning a struct might be good.

File: mythril/laser/ethereum/instructions.py (hunk 4 above)
Comment (nbanmp) [reply mythril/laser/ethereum/instructions.py]: It's checked in the above if.

File: mythril/laser/ethereum/instructions.py (hunk 3 above)
Comment (nbanmp) [reply mythril/laser/ethereum/instructions.py]: There's currently a `# TODO: maybe use BitVec here constrained to 1` at the bottom of `native_call()`, which is probably a better place to put this anyway.

File: mythril/laser/ethereum/instructions.py (hunk 2 above)
Comment (nbanmp) [reply mythril/laser/ethereum/instructions.py]: Yeah, `get_call_parameters` returns 7 parameters, and 8 were expected, so a `ValueError` was being raised and caught every time.

File: mythril/laser/ethereum/instructions.py (hunk 4 above)
Comment (JoranHonig) [reply mythril/laser/ethereum/instructions.py]: Ah I missed that

File: mythril/laser/ethereum/instructions.py (hunk 1 above)
Comment (JoranHonig) [reply mythril/laser/ethereum/instructions.py]: Why did you resolve this?

Comment (nbanmp) [reply mythril/laser/ethereum/instructions.py]: Because I added `return [global_state]` to these catches and fixed lines 1090, 1066, and 1067.
